pkgs/applications/version-management/sapling/default.nix

```diff
@@ -0,0 +1,166 @@
+    sha256 = "sha256-vrSAKpzruRROmQhu/3A6ZCoT1qAFKSAAOiMPMpS754Q=";
+  };
+  prompt_toolkit = fetchurl {
+    url = "https://files.pythonhosted.org/packages/87/61/2dfea88583d5454e3a64f9308a686071d58d59a55db638268a6413e1eb6d/prompt_toolkit-2.0.10-py3-none-any.whl";
+    sha256 = "sha256-RmQjRM5FdkHyj8nRycqTm2Pa343xKLhvG5hg5ZxzpeQ=";
+  };
+  decorator = fetchurl {
+    url = "https://files.pythonhosted.org/packages/6a/36/b1b9bfdf28690ae01d9ca0aa5b0d07cb4448ac65fb91dc7e2d094e3d992f/decorator-5.0.9-py3-none-any.whl";
+    sha256 = "sha256-blwZnBb3qfDjphpKVLPSfn2tDb3ekrlEQmyyCRQ3YyM=";
+  };
+  pickleshare = fetchurl {
+    url = "https://files.pythonhosted.org/packages/9a/41/220f49aaea88bc6fa6cba8d05ecf24676326156c23b991e80b3f2fc24c77/pickleshare-0.7.5-py2.py3-none-any.whl";
+    sha256 = "sha256-lkmvQU101N8RXV1xj4KstZydQYGWt7QpDtR6Es5i31Y=";
+  };
+  Pygments = fetchurl {
+    url = "https://files.pythonhosted.org/packages/a6/c9/be11fce9810793676017f79ffab3c6cb18575844a6c7b8d4ed92f95de604/Pygments-2.9.0-py3-none-any.whl";
+    sha256 = "sha256-1m6ARBEnhZTXZPxp7DbsE9mukUcZOhdAzTTScso4O44=";
+  };
+  traitlets = fetchurl {
+    url = "https://files.pythonhosted.org/packages/ca/ab/872a23e29cec3cf2594af7e857f18b687ad21039c1f9b922fac5b9b142d5/traitlets-4.3.3-py2.py3-none-any.whl";
+    sha256 = "sha256-cLTGodkBnXtPaEaDIoj4aZiqO5IHxoIfNXimpqRn/kQ=";
+  };
+  six = fetchurl {
+    url = "https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl";
+    sha256 = "sha256-irsvHYaJCi37mJ+ad8/P0+R8KjVLAREXcTJviqJuAlQ=";
+  };
+  ipython_genutils = fetchurl {
+    url = "https://files.pythonhosted.org/packages/fa/bc/9bd3b5c2b4774d5f33b2d544f1460be9df7df2fe42f352135381c347c69a/ipython_genutils-0.2.0-py2.py3-none-any.whl";
+    sha256 = "sha256-ct03IzeZ5hlmbJ9jmp2oPDQBOnPou8eaemNI2Txh+rg=";
+  };
+  ipdb = fetchurl {
+    url = "https://files.pythonhosted.org/packages/fc/56/9f67dcd4a4b9960373173a31be1b8c47fe351a1c9385677a7bdd82810e57/ipdb-0.13.9.tar.gz";
+    sha256 = "sha256-lRvZpkcxxET9kHpc4mhUMCAIaml/a+CPfMLJp1KieMU=";
+  };
+  Cython = fetchurl {
+    url = "https://files.pythonhosted.org/packages/bc/fa/8604d92ef753e0036d807f1b3179813ab2fa283e3b19c926e11673c8205b/Cython-0.29.26.tar.gz";
+    sha256 = "sha256-rzd9VDp2KGfaEfz25Vj3pKU1/4aT8wzOEj+rEMAPoxI=";
+  };
+in
+python3Packages.buildPythonPackage rec {
+  pname = "sapling";
+  version = "0.1.20221118-210929-cfbb68aa";
+
+  src = fetchFromGitHub {
+    owner = "facebook";
+    repo = pname;
+    rev = version;
+    hash = "sha256-IzbUaFrsSMojhsbpnRj1XLkhO9V2zYdmmZls4mtZquw=";
+  };
+
+  sourceRoot = "source/eden/scm";
+
+  # Upstream does not commit Cargo.lock, which buildRustPackage needs.
+  # postPatch is needed because lockFileContents doesn't copy the lockfile
+  # to src.
```

Comment:
```suggestion
  # Upstream does not commit Cargo.lock
```

Reply:
> There is no crate by chance so we can avoid vendoring the cargo.lock?

I haven't been able to avoid needing `Cargo.lock`; it seems to be needed for fetchCargoTarball.

Reply:
Might've interpreted that wrong; there is no sapling crate pushed to crates.io either